File: easy/0700_search_in_a_binary_search_tree/solution.cpp
#include "../../utils/cpp/nodes.hpp"

class Solution {
public:
  utils::TreeNode *searchBST(utils::TreeNode *root, int val) {
    while (root) {
      if (root->val == val)
        return root;
      else if (root->val < val)
        root = root->right;
      else
        root = root->left;
    }

    return nullptr;
  }
};

File: easy/0700_search_in_a_binary_search_tree/Solution.java
import nodes.TreeNode;

class Solution {
  public TreeNode searchBST(TreeNode root, int val) {
    TreeNode subroot = null;
    TreeNode curr = root;

    while (curr != null) {
      if (curr.val == val) {
        subroot = curr;
        break;
      } else if (curr.val < val) {
        curr = curr.right;
      } else {
        curr = curr.left;
      }
    }

    return subroot;
  }
}
